camel-core/src/main/java/org/apache/camel/processor/Throttler.java

```diff
@@ -61,25 +70,32 @@
 
     private static final String PROPERTY_EXCHANGE_QUEUED_TIMESTAMP = "CamelThrottlerExchangeQueuedTimestamp";
     private static final String PROPERTY_EXCHANGE_STATE = "CamelThrottlerExchangeState";
+    // (throttling grouping) defaulted as 1 because there will be only one queue which is similar to implementation
+    // when there is no grouping for throttling
+    private static final Integer NO_CORRELATION_QUEUE_ID = new Integer(1);
```

Comment:
Does it mean that we mix the default group when a correlationExpression evaluates to 1 ?

Reply:
Yes, this is correct. Not sure if that would happen very easily. Could not come up with something brighter. Open to hear ideas.

Reply:
Maybe keep throttleRate/delayQueue for default group and throttleRatesMap/defaultQueueCache for other groups.
Perhaps, put that in a dedicated class named `ThrottleRates` ?
Another idea could be to use a less common value, e.g. `NO_CORRELATION_QUEUE_ID = new Integer(Integer.MAX_VALUE)`.
Note that having a `correlationExpression` that could evaluate to the default group key would allow dynamic inclusion in the default group.

Reply:
nice comments, addressed some of them.
considering the ones(till others) left undone